omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/helper/LayoutTemplateProvider.java

```diff
@@ -48,86 +38,79 @@ public LayoutTemplateProvider(LayoutSupport<T> layoutSupport, String layoutDefau
-		description.addProperty("displayName", rep);
-		description.addProperty("codeName", rep);
-		description.addProperty("country", rep);
-		description.addProperty("lines", rep);
-		description.addProperty("lineByLineFormat", rep);
-		description.addProperty("nameMappings", rep);
-		description.addProperty("sizeMappings", rep);
-		description.addProperty("elementDefaults", rep);
-		description.addProperty("elementRegex", rep);
-		description.addProperty("elementRegexFormats", rep);
-		description.addProperty("requiredElements", rep);
-		return description;
-	}
-	
-	public ModelImpl getGETModel(Class<? extends Enum<?>> clsTokenEnum) {
-		return new ModelImpl()
-			.property("displayName", new StringProperty())
-			.property("codeName", new StringProperty())
-			.property("country", new StringProperty())
-			.property("lines", new ArrayProperty(
-					new ArrayProperty(
-							new ObjectProperty()
-									.property("isToken", new EnumProperty(clsTokenEnum))
-									.property("displayText", new StringProperty())
-									.property("codeName", new StringProperty())
-									.property("displaySize", new StringProperty())
-					)))
-			.property("lineByLineFormat", new ArrayProperty(new StringProperty()))
-			.property("nameMappings", new MapProperty().additionalProperties(new StringProperty()))
-			.property("sizeMappings", new MapProperty().additionalProperties(new StringProperty()))
-			.property("elementDefaults", new MapProperty().additionalProperties(new StringProperty()))
-			.property("elementRegex", new MapProperty().additionalProperties(new StringProperty()))
-			.property("elementRegexFormats", new MapProperty().additionalProperties(new StringProperty()))
-			.property("requiredElements", new ArrayProperty(new StringProperty()));
+	private T populateLayoutTemplate(T template) {
+		if (template == null) {
+			return null;
+		}
+		Locale locale = Context.getLocale();
+		T translated = translateNameMappings(template, locale);
+		return populateTemplateDefaults(translated, locale);
 	}
 	
-	public SimpleObject asRepresentation(T instance) throws ConversionException {
-		List<List<Map<String, String>>> lines = getLines(instance);
-		SimpleObject obj = new SimpleObject();
-		obj.add("displayName", instance.getDisplayName());
-		obj.add("codeName", instance.getCodeName());
-		obj.add("country", instance.getCountry());
-		obj.add("lines", lines);
-		obj.add("lineByLineFormat", instance.getLineByLineFormat());
-		obj.add("nameMappings", instance.getNameMappings());
-		obj.add("sizeMappings", instance.getSizeMappings());
-		obj.add("elementDefaults", instance.getElementDefaults());
-		obj.add("elementRegex", instance.getElementRegex());
-		obj.add("elementRegexFormats", instance.getElementRegexFormats());
-		obj.add("requiredElements", instance.getRequiredElements());
-		return obj;
+	private T translateNameMappings(T template, Locale locale) {
+		MessageSourceService messages = Context.getMessageSourceService();
```

Comment:
Is there a reason why you are calling this `messages` instead of something like `messageService`

Reply:
Renamed.